File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/wallet/WalletDao.java
package com.github.dedis.popstellar.repository.database.wallet;

import androidx.room.*;

import io.reactivex.Completable;

/** Interface to query the table containing the wallet seed */
@Dao
public interface WalletDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  Completable insert(WalletEntity walletEntity);

  @Query("SELECT * FROM wallet LIMIT 1")
  WalletEntity getWallet();
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/wallet/WalletDao.kt
package com.github.dedis.popstellar.repository.database.wallet

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Completable

/** Interface to query the table containing the wallet seed */
@Dao
interface WalletDao {
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  fun insert(walletEntity: WalletEntity): Completable

  @get:Query("SELECT * FROM wallet LIMIT 1") val wallet: WalletEntity?
}
